src/avram/charms/time_extensions.cr

```diff
@@ -3,16 +3,51 @@ struct Time
     alias ColumnType = Time
     include Avram::Type
 
+    TIME_FORMATS = [
+      Time::Format::ISO_8601_DATE,
+      Time::Format::ISO_8601_DATE_TIME,
+      Time::Format::ISO_8601_TIME,
+      Time::Format::RFC_2822,
+      Time::Format::RFC_3339,
+      Time::Format::HTTP_DATE,
+    ]
+
     def self.from_db!(value : Time)
       value
     end
 
-    def self.parse(value : String)
-      SuccessfulCast(Time).new Time.parse_iso8601(value).to_utc
+    def self.parse(value : String) : SuccessfulCast(Time) | FailedCast
+      try_parsing_with_default_formatters(value) ||
+        try_parsing_with_string_formats(value) ||
+        FailedCast.new
     rescue Time::Format::Error
       FailedCast.new
     end
 
+    def self.try_parsing_with_default_formatters(value : String)
+      TIME_FORMATS.find do |format|
+        begin
+          format.parse(value)
+        rescue e : Time::Format::Error
+          nil
+        end
+      end.try do |format|
+        SuccessfulCast.new format.parse(value).to_utc
+      end
+    end
+
+    def self.try_parsing_with_string_formats(value)
+      Avram.settings.time_formats.find do |format|
+        begin
+          Time.parse(value, format, Time::Location.load("UTC"))
+        rescue e : Time::Format::Error
```

Comment:
I wish Crystal had some way to return `nil`  but it only raises when parsing so we have to do this kind of hacky begin/rescue dance :(

Reply:
Working with Time in crystal is actually really clunky.